plasmapy/atomic/particle_class.py

```diff
@@ -480,6 +501,35 @@ def __bool__(self):
         """
         raise AtomicError("The truthiness of a Particle object is not defined.")
 
+    def __invert__(self):
+        """
+        Return the corresponding antiparticle, or raise an
+        `~plasmapy.utils.AtomicError` if the particle is not an
+        elementary particle.
+        """
+        if self.element and self != 'p+':
+            raise AtomicError(
+                "The unary operator can only be used for elementary particles and antiparticles.")
+
+        old_particle = self.particle
+
+        if old_particle == 'n':
+            return Particle('antineutron')
+
+        elif '-' in old_particle:
+            return Particle(old_particle.replace('-', '+'))
+        elif '+' in old_particle:
+            return Particle(old_particle.replace('+', '-'))
+        elif 'anti_' in old_particle:
+            return Particle(old_particle.replace('anti_', ''))
+        elif 'anti' in old_particle:
+            return Particle(old_particle.replace('anti', ''))
+        else:
+            try:
+                return Particle('anti' + old_particle)
+            except InvalidParticleError:
+                return Particle('anti_' + old_particle)
+
```

Comment:
I think I made this more complicated than I needed it to be.  I am going create a dict (or maybe a defaultdict) that has particles as keys and their antiparticles as the corresponding values.